test/jdk/com/sun/jdi/JdwpNetProps.java

```diff
@@ -60,33 +61,82 @@ public static void main(String[] args) throws Exception {
             }
         }
 
+        String preferIPv6Address = System.getProperty("java.net.preferIPv6Addresses");
+        if (!Objects.equals(preferIPv6Address, "system")) {
+          throw new AssertionError(
+              "Expected -Djava.net.preferIPv6Address=system, was " + preferIPv6Address);
+        }
+        boolean systemPrefersIPv6 = addrs[0] instanceof Inet6Address;
+
         if (ipv4Address != null) {
             new ListenTest("localhost", ipv4Address)
                     .preferIPv4Stack(true)
                     .run(TestResult.Success);
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(true)
+                    .preferIPv6Addresses("true")
+                    .run(TestResult.Success);
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(true)
+                    .preferIPv6Addresses("system")
+                    .run(TestResult.Success);
             new ListenTest("localhost", ipv4Address)
                     .preferIPv4Stack(false)
                     .run(TestResult.Success);
+            new ListenTest("localhost", ipv4Address)
+                    .preferIPv4Stack(false)
+                    .preferIPv6Addresses("true")
+                    .run(TestResult.AttachFailed);
```

Comment:
If test system is IPv4-only, this testcase will success
So it should be under "ipv6Address != null" section

Reply:
Thanks, done

I had been testing on dual-stack on IPv6-only, which missed that. I confirmed I could repro the expected failure with IPv4-only.